transformer_engine/pytorch/cpp_extensions/gemm.py

```diff
@@ -109,6 +110,21 @@ def general_gemm(
         if not out.is_contiguous():
             raise ValueError("Output tensor is not contiguous.")
 
+    debug_quantizer = None
+    if isinstance(quantization_params, DebugQuantizer):
+        debug_quantizer = quantization_params
+        quantization_params = quantization_params.parent_quantizer
+        A = A.get_tensor(not transa)
+        B = B.get_tensor(transb)
+
+        assert (type(A) in [torch.Tensor, torch.nn.parameter.Parameter]) == (
+            type(B) in [torch.Tensor, torch.nn.parameter.Parameter]
+        ), (
+            "[Debug tools] Processed tensors should both be FP8 tensors or both be torch tensors  "
+            f"            but type(A) = {type(A)},             "
+            f"  type(B) = {type(B)}"
+        )
```

Comment:
```suggestion
        assert isinstance(A, torch.Tensor) == isinstance(B, torch.Tensor), (
            "[Debug tools] Processed tensors should both be FP8 tensors or both be torch tensors  "
            f"            but type(A) = {type(A)},             "
            f"  type(B) = {type(B)}"
        )
```

Reply:
QuantizedTensor is a subclass of torch.Tensor so it would pass yield `True` in `isinstance(A, torch.Tensor)`. See my previous comment.